File: Pascal/VCL/VCLFormUnit.pas
unit VCLFormUnit;

interface

uses
  Winapi.Windows, Winapi.Messages, System.SysUtils, System.Variants, System.Classes, Vcl.Graphics,
  Vcl.Controls, Vcl.Forms, Vcl.Dialogs, Vcl.CastleControl,
  Vcl.ExtCtrls, Vcl.StdCtrls, Vcl.ComCtrls,
  CastleAppUnit
  ;

type
  { TForm }
  TForm1 = class(TForm)
    CastleControl1: TCastleControl;
    Panel1: TPanel;
    RadioGroup1: TRadioGroup;
    TrackBar1: TTrackBar;
    Label1: TLabel;
    procedure FormCreate(Sender: TObject);
    procedure RadioGroup1Click(Sender: TObject);
    procedure TrackBar1Change(Sender: TObject);
  private
    { Private declarations }
    CastleApp: TCastleApp;
  public
    { Public declarations }
  end;

var
  Form1: TForm1;

implementation

{$R *.dfm}

procedure TForm1.FormCreate(Sender: TObject);
begin
  CastleControl1.Align := TAlign.alClient;
  CastleControl1.Parent := Self;
  CastleApp := TCastleApp.Create(CastleControl1);
  CastleControl1.Container.View := CastleApp;

  RadioGroup1.Items.Add('2D');
  RadioGroup1.Items.Add('3D');
  RadioGroup1.ItemIndex := 1;
end;

procedure TForm1.RadioGroup1Click(Sender: TObject);
begin
  if RadioGroup1.ItemIndex = 0 then
    CastleApp.SwitchView3D(False)
  else
    CastleApp.SwitchView3D(True);
end;

procedure TForm1.TrackBar1Change(Sender: TObject);
begin
  if Assigned(CastleApp.ActiveScene) then
    begin
      Label1.Caption := 'Rotation : ' + IntToStr(Trackbar1.Position);
      CastleApp.SetRotation(Trackbar1.Position);
    end;
end;

end.

File: CPP/VCL/__astcache/c@@git@cgeformexamples@cpp@vcl/vclformunit.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "VCLFormUnit.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "Vcl.CastleControl"
#pragma resource "*.dfm"
TForm1 *Form1;
//---------------------------------------------------------------------------
__fastcall TForm1::TForm1(TComponent* Owner)
	: TForm(Owner)
{
}
//---------------------------------------------------------------------------

void __fastcall TForm1::FormCreate(TObject *Sender)
{
  CastleControl1->Align = alClient;
  CastleControl1->Parent = this;
  CastleApp = new TCastleApp(CastleControl1);
  CastleControl1->Container->View = CastleApp;

  RadioGroup1->Items->Add("2D");
  RadioGroup1->Items->Add("3D");
  RadioGroup1->ItemIndex = 1;
  RadioGroup1->Columns = 2;
}
//---------------------------------------------------------------------------

void __fastcall TForm1::RadioGroup1Click(TObject *Sender)
{
  if(RadioGroup1->ItemIndex == 0)
  {
	CastleApp->SwitchView3D(False);
  }
  else
  {
	CastleApp->SwitchView3D(True);
  }
}
//---------------------------------------------------------------------------

void __fastcall TForm1::TrackBar1Change(TObject *Sender)
{
  Label1->Caption = "Rotation : " + IntToStr(TrackBar1->Position);
  CastleApp->SetRotation(TrackBar1->Position);
}
//---------------------------------------------------------------------------

